### PR TITLE
stdin: on HUP, read everything

### PR DESCRIPTION
**Pull request by nalind**
When we're polling to handle stdio for a container, when we detect a HUP on our stdin, read all that we can from stdin before closing it, instead of reading only, at most, a single chunk of bytes.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@TomSweeneyRedHat @pixdrift @vrothberg @umohnani8 PTAL

**Comment by vrothberg** [comment]
Besides the two logs, LGTM :+1:

**Comment by TomSweeneyRedHat** [comment]
LGTM once questions are answered or addressed.

File: chroot/run.go [hunk 1]
```diff
@@ -381,11 +381,15 @@ func runUsingChrootMain() {
...
 		return false
 	}
-	for readFd := range relays {
+	for readFd, writeFd := range relays {
 		if err := unix.SetNonblock(readFd, true); err != nil {
 			logrus.Errorf("error setting descriptor %d (%s) non-blocking: %v", readFd, fdDesc[readFd], err)
 			return
 		}
+		if err := unix.SetNonblock(writeFd, false); err != nil {
+			logrus.Errorf("error setting descriptor %d (%s) blocking: %v", relays[writeFd], fdDesc[writeFd], err)
```
[the remark this answers is not in the document]
Comment (nalind) [reply chroot/run.go]: @TomSweeneyRedHat yep, you're reading that right.

**Comment by nalind** [comment]
Rebased.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**